src/Generic.Repository/Models/PageAggregation/PageAbstract.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Generic.Repository.Cache;
+using Generic.Repository.Models.PageAggregation.PageConfig;
+using Generic.Repository.ThrowError;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Generic.Repository.Models.PageAggregation
+{
+    public abstract class PageAbstract<TValue> : PageAttrAbstract<TValue, TValue>
+    where TValue : class
+    {
+        protected PageAbstract(
+            ICacheRepository cacheRepository,
+            IQueryable<TValue> listEntities,
+            IPageConfig config
+        ) : base(listEntities, config, cacheRepository) { }
+    }
+
+    public abstract class PageAbstract<TValue, TResult> : PageAttrAbstract<TValue, TResult>
+        where TValue : class
+        where TResult : class
+    {
+        #region CONST AND READONLY
+
+        /// <summary>The mapping</summary>
+        protected readonly Func<IEnumerable<object>, IEnumerable<TResult>> Mapping;
+
+        #endregion CONST AND READONLY
+
+        #region CTOR
+
+        public PageAbstract(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Change the visibility of this constructor to 'protected'.](https://app.codacy.com/manual/guilhermecaixeta/Generic.Service.DotNetCore2.2/pullRequest?prid=4961311)